paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/FormFieldValues.kt

```diff
@@ -1,12 +1,19 @@
 package com.stripe.android.paymentsheet.forms
 
+import com.stripe.android.paymentsheet.ElementType
 import com.stripe.android.paymentsheet.specifications.IdentifierSpec
 
 /**
  * The identifier here comes from the form element (section, static text, etc)
  */
 class FormFieldValues(
-    val fieldValuePairs: Map<IdentifierSpec, String?> = mapOf(),
+    val fieldValuePairs: Map<IdentifierSpec, FormFieldEntry> = mapOf(),
     val saveForFutureUse: Boolean,
     val showsMandate: Boolean
 )
+
+data class FormFieldEntry(
+    val type: ElementType,
```

Comment:
Since `ElementType` is an type, should it be the key of fieldValuePairs?

Reply:
In this case we don't because ultimately we are using the Identifier in the fieldValuePairs (the key) as a key into the PaymentMethodCreateParams map.  Similarly the the identifier is used to associate the formFieldValues with the elements in the Form when repopulating it.

Let me know if it would help to discuss further.